### ISSUE TITLE
Reset UserId while logged out

### ISSUE THREAD

**Issue by JeyakumarThangaraj** [matomo-org/matomo-sdk-android#259]
Team,

We dont find anyway to reset userId while logout. because of this, visitor id remain same for all the time whenever same user is trying to login. Kindly Advise

Thanks,
Jeyakumar

**Comment by d4rken** [comment]
What about `Tracker.setUserId(...)`.

https://github.com/matomo-org/matomo-sdk-android/blob/006ed7dfb4541273adc43bbd315bcd955dc32379/tracker/src/main/java/org/matomo/sdk/Tracker.java#L334

**Comment by JeyakumarThangaraj** [comment]
If i reset user Id, new visitor ID and a new visit is not created. Still existing visit only being used. Kindly advise. 
I want below to be implemented in android.

![image](https://user-images.githubusercontent.com/49137580/60824687-b38b1d00-a1b2-11e9-9ce6-9d052356f7b7.png)

**Comment by d4rken** [comment]
`resetUserId` also resets the visitor-id in the [javascript tracker](https://github.com/matomo-org/matomo/blob/7a93a2b5ff683e7bb27339502a60c00add9fe26e/tests/javascript/index.php)

So to do the analog here you'd probably have to do something like:

```kotlin
tracker.setUserId(null)
tracker.setVisitorId(Tracker.makeRandomVisitorId())
```

Also see `SESSION_START` (i.e. [`new_visit`](https://developer.matomo.org/api-reference/tracking-api)), which you could set to prevent the server from attempting any visit stitching.

```kotlin
TrackHelper.track(TrackMe().set(QueryParams.SESSION_START,1))
    .screen("/").title("Main screen")
    .with(getTracker())
```

**Comment by Nithiyasabariswaran** [comment]
Hi All,

All my screens are loading in a single visitor id (Android).
I have used tracker.setVisitorId(Tracker.makeRandomVisitorId()) also.

kinldy advice,
Thanks in advance.

![image](https://user-images.githubusercontent.com/12802349/60833760-afb5c580-a1c7-11e9-86e4-44ac102cbae3.png)

**Comment by mattab** [comment]
> resetUserId also resets the visitor-id in the javascript tracker

Fyi @d4rken we recently changed this and no longer will resetUserId change the Visitor ID. This was done in https://github.com/matomo-org/matomo/pull/14360

**Comment by hannesa2** [comment]
Solved with  #373
If not, please reopen it
